src/ahp/constraint_systems.rs

```diff
@@ -108,60 +80,84 @@ pub(crate) fn make_matrices_square<F: Field>(
     }
 }
 
+/// Evaluations of various polynomials related to the constraint matrices,
+/// over the same domain.
 #[derive(Derivative, CanonicalSerialize, CanonicalDeserialize)]
-#[derivative(Clone(bound = "F: PrimeField"))]
+#[derivative(Debug(bound = "F: PrimeField"), Clone(bound = "F: PrimeField"))]
 pub struct MatrixEvals<F: PrimeField> {
```

Comment:
We jointly arithmetize all the matrices at once, so now I store `val_a`, `val_b`, and `val_c` all in the same place.